data/tutorials/getting-started/02_fst_program.md

```diff
@@ -0,0 +1,310 @@
+```ocaml
+# #show Hello;;
+module Hello : sig val world : string end
+```
+
+## Defining Module Interfaces
+
+What UTop's `#show` command displays is an API, the list of definitions provided by a module, which is called a _module interface_. In a similar way, module can be defined by `.ml` files. It is also possible to define module interfaces from files. Such files have a `.mli` extension and must share the same name part. 
+```shell
+$ echo 'val world : string' > lib/hello.mli
+```
+
+Observe that only what is between `sig` and `end` needs to written in the interface file. This is explained in the tutorial dedicated to [modules](link_modules).
+
+Module interfaces are also used to create _private_ definitions. A module definition is private if it is not listed in its corresponding interface. If no interface file exists, everyting is public.
+
+Edit the `lib/hello.ml` file to add the `mundo` definition, as shown below:
+```ocaml
+let mundo = "¡Hola Mundo!"
+let world = mundo
+```
+
+Also edit the `bin/main.ml` file like this:
+```ocaml 
+let () = Printf.printf "%s\n" Hello.mundo
+```
+
+Trying to compile this fails.
+```shell
+$ dune build
+Entering directory '/home/cuihtlauac/caml/ocaml.org'
+File "hello/bin/main.ml", line 1, characters 30-41:
+1 | let () = Printf.printf "%s\n" Hello.mundo
+                                  ^^^^^^^^^^^
+Error: Unbound value Hello.mundo
+```
+
+This is because we haven't changed `hello.mli`. Since it does not list `mundo`, it is therefore private.
+
+## Installing and Using Modules from a Package
+
+OCaml has an active community of open-source contributors. Most projects are avaiable using the opam package manager, which you installed in the "Install OCaml" tutorial. The following section shows how to install and use a package from opam's open-source repository.
+
+To illustrate this, let's turn our modest `hello` project into a web server using Anton Bachin's Dream web framework. First install the `dream` package with this command:
+```shell
+$ opam install dream
+```
+
+Next, call the framework in the `main.ml` file:
+```ocaml
+let () = Dream.(run (router [ get "/" (fun _ -> html Hello.world) ]))
+```
+
+This is telling: “run a web server responding to `/` requests with `Hello.world` as if it was HTML content.” The `Dream.(` syntax stands for locally opening a module inside an expression.
+
+Finally, tell Dune it is going to need Dream to compile the project. Do this by just adding `dream` in the `library` stanza of the `bin/dune` file. Here's how:
+```lisp 
+(executable
+ (public_name hello)
+ (name main)
+ (libraries hello dream))
+```
+
+Launch the server from a new terminal.
+```shell
+$ dune exec hello
+20.07.23 13:14:07.801                       0
+20.07.23 13:14:07.801                       Type Ctrl+C to stop
+```
+
+Then test from another:
+```shell
+$ curl http://localhost:8080/
+¡Hola Mundo!
+```
```

Comment:
Get these errors:
```
christinerose@Christines-MacBook-Air bin % dune exec hello
Entering directory '/Users/christinerose/Documents/Christine/Career/Tarides/GitHub_Repos/hello'
Error: Multiple rules generated for _build/install/default/bin/hello:
- bin/dune:8
- bin/dune:3
christinerose@Christines-MacBook-Air bin % curl http://localhost:8080/
curl: (7) Failed to connect to localhost port 8080 after 5 ms: Couldn't connect to server
christinerose@Christines-MacBook-Air bin % 
```

Reply:
Still get the same errors, even after the update. Perhaps we'll need to go through this together, @cuihtlauac, as either I'm doing something wrong and/or it needs some clarification.

Reply:
Worked after fixing the `dune` file

Reply:
Add note about seeing the password sign in macOS. Doesn't mean anything, just type the Ctrl+C to get out of that.